flow/Platform.actor.cpp

```diff
@@ -2752,7 +2752,24 @@ THREAD_HANDLE startThread(void* (*func)(void*), void* arg, int stackSize, const
 #if defined(__linux__)
 	if (name != nullptr) {
 		// TODO: Should this just truncate?
-		ASSERT_EQ(pthread_setname_np(t, name), 0);
+		int retVal = pthread_setname_np(t, name);
+		if (!retVal)
+			return t;
+		// In simulation and unit testing a thread may return before the name can be set, this will
+		// return ENOENT or ESRCH. We'll log when ENOENT or ESRCH is encountered and continue, otherwise we'll log and
+		// throw a platform_error.
+		if (errno == ENOENT || errno == ESRCH) {
+			TraceEvent(SevWarn, "PthreadSetNameNp")
+			    .detail("Name", name)
+			    .detail("ReturnCode", retVal)
+			    .detail("Errno", errno);
```

Comment:
For trace events, we have a `GetLastError` method that you can use to log `errno` in a standard way. You can see other examples of this elsewhere in Platform.actor.cpp.